src/frontend/js/components/AshleyEditor/LatexButton/index.tsx

```diff
@@ -9,17 +9,62 @@ import { FormattedMessage } from 'react-intl';
 import Tooltip from 'rc-tooltip';
 import { messagesEditor } from '../messages';
 import { DraftHandleValue, TypeLatexStyle } from '../../../types/Enums';
+
 interface LatexButtonProps {
   type: string;
   theme: DraftJsButtonTheme;
+  editorState: EditorState;
   getEditorState: () => EditorState;
   setEditorState: (editorState: EditorState) => void;
 }
 
 export const LatexButton = (props: LatexButtonProps) => {
   const node = useRef<HTMLDivElement>(null);
-  const { buttonWrapper = '', button } = props.theme;
-  const imageClassName = useMemo(() => `latex-${props.type}`, [props.type]);
+  const { buttonWrapper = '', button, active } = props.theme;
+  const [isActive, setIsActive] = useState(false);
+
+  const isInlineTeXSelected = () => {
+    const selection = props.editorState.getSelection();
+    const contentState = props.editorState.getCurrentContent();
+    const blockKey = selection.getStartKey();
+    const block = contentState.getBlockForKey(blockKey);
+    if (!block) {
+      return false;
+    }
+    const offset = selection.getIsBackward()
+      ? selection.getStartOffset()
+      : selection.getEndOffset();
+    const entityKey = block.getEntityAt(offset);
+    if (
+      entityKey &&
+      contentState.getEntity(entityKey).getType() === 'INLINETEX'
+    ) {
+      return true;
+    }
+    return false;
```

Comment:
Could be simplified : 

```suggestion
    const selection = props.editorState.getSelection();
    const contentState = props.editorState.getCurrentContent();
    const blockKey = selection.getStartKey();
    const block = contentState.getBlockForKey(blockKey);
    const offset = selection.getIsBackward()
          ? selection.getStartOffset()
          : selection.getEndOffset();
    const entityKey = block?.getEntityAt(offset);

    if (!entityKey) {
        return false;
    }

    return contentState.getEntity(entityKey).getType() === 'INLINETEX';
```